src/components/CurrentImage/index.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from "react";
+import { Col, Row } from "antd";
+import puzzledImage from "./../../images/habitat.jpg";
+import "./index.css";
+
+// CurrentImage is the component that generates and includes the tiles of the puzzle
+// it generates 6 tiles ordered vertically on top of each other
+// each tile's background is 1/6 of the image. After the tiles are generated they are put
+// in an array and ordered randomly
+// a user can click on one tile to select and then click on another tile to swap their places
+// the user will swap the places of each tile to put it in its correct position
+// when all tiles are placed in their correct position the full image is displayed
+const CurrentImage = () => {
+  const [generatedTiles, setGeneratedTiles] = useState([]);
+  // selectedTile state shows if there is a tile selected and which one is selected if any
+  const [selectedTile, setSelectedTile] = useState(null);
+  // if unsolved it displays the tiles and when solved it displays the full image
+  const [isSolved, setIsSolved] = useState(false);
+  const tilesGenerator = () => {
+    const newTilesArray = [];
+    // for loop to generate tiles where tiles are puzzle pieces
+    const numOfTiles = 6;
+    for (let tileId = 0; tileId < numOfTiles; tileId++) {
+      let tileTopStyling = tileId * 100;
+      // value of selected is used for styling purposes where the selected tile should be highlighted
+      // value of top is used for fitting the image inside the tile
+      let newTile = { id: tileId, selected: false, top: tileTopStyling };
+      newTilesArray.push(newTile);
+    }
+    // for loop that generates a random and extracts the element with the index
+    // equal to the random number, then pushes that element to the end of newTilesArray.
+    for (let i = 0; i < newTilesArray.length; i++) {
+      let random = Math.floor(Math.random() * (newTilesArray.length - 1));
+      let randomItemArray = newTilesArray.splice(random, 1);
+      newTilesArray.push(randomItemArray[0]);
+    }
+    // Set the randomized array as generated tiles
+    setGeneratedTiles(newTilesArray);
```

Comment:
newTilesArray should probably just be called randomizedTilesArray

Reply:
I know you said I should rename it, but instead I thought it's better to create a new array with the same values of newTilesArray but randomized, using the shuffle function Tarkan created. PTAL